### PR TITLE
Closes #3788: Expose to which category a blocked tracker belongs

### LINKED ISSUE #3788: Expose to which category a blocked tracker belongs to
We are exposing the [URLs of the blocked trackers](https://github.com/mozilla-mobile/android-components/blob/1d62cefa466bff858587804b99991496edd97ae5/components/browser/session/src/main/java/mozilla/components/browser/session/Session.kt#L305) but we also need to indicate to which category they belong to. 

Related issue https://github.com/mozilla-mobile/fenix/issues/3571

### PR DESCRIPTION
**Pull request by Amejia481**
### Pull Request checklist

- [x] **Quality**: This PR builds and passes detekt/ktlint checks (A pre-push hook is recommended)
- [x] **Tests**: This PR includes thorough tests or an explanation of why it does not
- [x] **Changelog**: This PR includes [a changelog entry](https://github.com/mozilla-mobile/android-components/blob/master/docs/changelog.md) or does not need one
- [ ] **Accessibility**: The code in this PR follows [accessibility best practices](https://github.com/mozilla-mobile/shared-docs/blob/master/android/accessibility_guide.md) or does not include any user facing features

### After merge
- [x] **Milestone**: Make sure issues closed by this pull request are added to the [milestone](https://github.com/mozilla-mobile/android-components/milestones) of the version currently in development.

### CODE REVIEW

File: components/concept/engine/src/main/java/mozilla/components/concept/engine/content/blocking/Tracker.kt [hunk 1]
```diff
@@ -0,0 +1,13 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+package mozilla.components.concept.engine.content.blocking
```
Comment (csadilek) [inline components/concept/engine/src/main/java/mozilla/components/concept/engine/content/blocking/Tracker.kt]: nit: new line

File: components/concept/engine/src/main/java/mozilla/components/concept/engine/content/blocking/Tracker.kt [hunk 2]
```diff
@@ -0,0 +1,13 @@
...
+/**
+ * Represents a blocked content.
```
Comment (csadilek) [inline components/concept/engine/src/main/java/mozilla/components/concept/engine/content/blocking/Tracker.kt]: nit: a blocked content tracker? Let's also document the params.

File: docs/changelog.md [hunk 3]
```diff
@@ -11,6 +11,10 @@ permalink: /changelog/
 * [Gecko](https://github.com/mozilla-mobile/android-components/blob/master/buildSrc/src/main/java/Gecko.kt)
 * [Configuration](https://github.com/mozilla-mobile/android-components/blob/master/buildSrc/src/main/java/Config.kt)
 
+* ****concept-engine****, **browser-session**
```
Comment (csadilek) [inline docs/changelog.md]: nit: formating problem: `****concept-engine****`

File: docs/changelog.md [hunk 4]
```diff
@@ -11,6 +11,10 @@ permalink: /changelog/
...
+  * ⚠️ **This is a breaking change**: Function signature changed  `Session.Observer.onTrackerBlocked(session: Session, blocked: String, all: List<String>) = Unit` from to `Session.Observer.onTrackerBlocked(session: Session, tracker: Tracker, all: List<Tracker>) = Unit`
```
Comment (csadilek) [inline docs/changelog.md]: Let's maybe also add what we added e.g. that we now provide more details about a blocked content tracker, not just the changed method signature.

Comment (Amejia481) [reply docs/changelog.md]: 👍

Comment (Amejia481) [reply docs/changelog.md]: Added more info :)

File: docs/changelog.md (hunk 3 above)
Comment (Amejia481) [reply docs/changelog.md]: Fixed!

File: components/concept/engine/src/main/java/mozilla/components/concept/engine/content/blocking/Tracker.kt (hunk 2 above)
Comment (Amejia481) [reply components/concept/engine/src/main/java/mozilla/components/concept/engine/content/blocking/Tracker.kt]: Fixed!

File: components/concept/engine/src/main/java/mozilla/components/concept/engine/content/blocking/Tracker.kt (hunk 1 above)
Comment (Amejia481) [reply components/concept/engine/src/main/java/mozilla/components/concept/engine/content/blocking/Tracker.kt]: Fixed!

File: components/browser/session/src/main/java/mozilla/components/browser/session/Session.kt [hunk 5]
```diff
@@ -300,9 +301,9 @@ class Session(
     }
 
     /**
-     * List of URIs that have been blocked in this session.
+     * List of [Tracker] that have been blocked in this session.
```
[the remark this answers is not in the document]
Comment (Amejia481) [reply components/browser/session/src/main/java/mozilla/components/browser/session/Session.kt]: Fixed!

**Pull request merged**